go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,295 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	bsgrpc "google.golang.org/genproto/googleapis/bytestream"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var logStreamData = []byte("Hello World! This is large data to send.")
+
+type logStream struct {
+	logStreamID   string
+	logicalOffset int64
+	finalized     bool
+}
+
+func TestWriteBytesWithOffsetSuccess_LogStream(t *testing.T) {
+	tests := []struct {
+		description  string
+		ls           *logStream
+		data         []byte
+		dataPartsLen int
+		opts         []ByteStreamWriteOption
+		wantBytesLen int64
+	}{
+		{
+			description:  "valid data with offset 0",
+			ls:           &logStream{logicalOffset: 0},
+			data:         logStreamData,
+			opts:         []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptFinishWrite(false)},
+			dataPartsLen: 3,
+			wantBytesLen: int64(len(logStreamData)),
+		},
+		{
+			description:  "valid data with non-zero offset",
+			ls:           &logStream{logicalOffset: 4},
+			data:         logStreamData,
+			opts:         []ByteStreamWriteOption{ByteStreamOptOffset(4), ByteStreamOptFinishWrite(false)},
+			dataPartsLen: 3,
+			wantBytesLen: int64(len(logStreamData)),
+		},
+		{
+			description:  "one big chunk",
+			ls:           &logStream{logicalOffset: 0},
+			data:         logStreamData,
+			opts:         []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptFinishWrite(true)},
+			dataPartsLen: 1,
+			wantBytesLen: int64(len(logStreamData)),
+		},
+		{
+			description:  "empty data",
+			ls:           &logStream{logicalOffset: 0},
+			data:         []byte{},
+			opts:         []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptFinishWrite(false)},
+			dataPartsLen: 1,
+			wantBytesLen: 0,
+		},
+	}
+
+	b := newServer(t)
+	defer b.shutDown()
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			size := len(test.data)/test.dataPartsLen + 1
+			start := int(test.opts[0].(ByteStreamOptOffset))
```

Comment:
Just using function parameters rather than an opt struct will make this simpler as well.

Reply:
Done.